datashader/core.py

```diff
@@ -123,6 +129,12 @@ def __init__(self, plot_width=600, plot_height=600,
         self.x_axis = _axis_lookup[x_axis_type]
         self.y_axis = _axis_lookup[y_axis_type]
 
+        if x_axis_type=='log' and x_range and x_range[0]<=0:
+            raise ValueError("x_range must be >0 for a log x axis")
+        if y_axis_type=='log' and y_range and y_range[0]<=0:
+            raise ValueError("y_range must be >0 for a log y axis")
```

Comment:
Yes, that's how all the reductions are done as well. See: https://github.com/bokeh/datashader/blob/master/datashader/reductions.py#L133-L136.